### PR TITLE
check-sof-logger: catch ERROR in firmware traces and fail

### PR DESCRIPTION
**Pull request by marc-hb**
Add function find_logger_error() in case-lib so it can re-used by other
tests later.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
https://sof-ci.01.org/softestpr/PR799/build917/devicetest/?model=BDW_WSB_RT286&testcase=check-sof-logger has a very interesting:
```
 TIMESTAMP      (us)           DELTA  C# COMPONENT          LOCATION                      CONTENT	ktime=117.383s  @  2021-11-01 23:27:33 +0000 UTC
[  1852324251.812] (           0.000) c0 dma-trace             src/trace/dma-trace.c:422  INFO SHM: FW ABI 0x3014000 DBG ABI 0x5003000 tag v1.9-rc1-211-gb0bb21dc0ee6 src hash 0xa9c07289 (ldc hash 0xa9c07289)
[      324686.862] (      324686.875) c0 dma-trace             src/trace/dma-trace.c:605  ERROR dtrace_add_event(): number of dropped logs = 1
```

**Comment by marc-hb** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
> https://sof-ci.01.org/softestpr/PR799/build917/devicetest/?model=BDW_WSB_RT286&testcase=check-sof-logger has a very interesting:

https://sof-ci.01.org/softestpr/PR799/build16/devicetest has something similar on BYT (BDW is not in PR testing anywmore)

```
[   178559514.207] (           0.000) c0 dma-trace             src/trace/dma-trace.c:347  INFO SHM: FW ABI 0x3016000 DBG ABI 0x5003000 tags SOF:v2.0-rc1-813-g5cfc37040bbe src hash 0xb72c3013 (ldc hash 0xb72c3013)
[       27442.082] (       27442.082) c0 dma-trace             src/trace/dma-trace.c:654  ERROR dtrace_add_event(): number of dropped logs = 8
```

**Comment by ujfalusi** [comment]
@marc-hb, is it a recent issue or it has been there?
It is very likely to happen if the firmware is generating 'enough' trace messages up to the point when `sof-logger` starts and opens the sof/trace debugfs file to start consuming the dtrace messages.

If you do not start the `sof-logger` at all then yeah, you are going to drop messages, but is this an error?

**Comment by marc-hb** [comment]
Thanks @ujfalusi for the prompt review!

Interesting. So this reproduced systematically on BYT and BDW... because the buffer is smaller / too small? Or maybe because they have more interfaces and more to log.

So I can just downgrade this ERROR to a WARN, right?

**Comment by ujfalusi** [comment]
@marc-hb, I think it is WARN rather an ERR, yes (if you refer to the fw print about the dropped new entries. It only takes to not start sof-logger to eventually have them, which is fine?

It can explain 'blackouts' in logs

**Comment by marc-hb** [comment]
https://sof-ci.01.org/softestpr/PR799/build112/devicetest is all green. I recommend we merge this "as is" and if this error message become too frequent we can downgrade it.

File: test-case/check-sof-logger.sh [hunk 1]
```diff
@@ -256,8 +256,31 @@ main()
...
+
+    case "$platf" in
+        byt|bdw)
+            dlogw 'not looking for ERROR on BYT/BDW because of known DMA issues #4333 and others'
+            ERROR_scan=false
+            ;;
+    esac
+
+    if $ERROR_scan; then
+    for f in "${stdout_files[@]}"; do
```
Comment (keqiaozhang) [inline test-case/check-sof-logger.sh:271]: Why not indent?

Comment (marc-hb) [reply test-case/check-sof-logger.sh:271]: It was supposed to be temporary, I can indent.

Comment (marc-hb) [reply test-case/check-sof-logger.sh:261]: You pushed me to make more changes, thanks.

**Comment by marc-hb** [comment]
https://sof-ci.01.org/softestpr/PR799/build113/devicetest/ has one usual WHL UPX suspend/resume TIMEOUT, everything else is green.

**Review by keqiaozhang — APPROVED** [review]
LGTM

**Comment by marc-hb** [comment]
Thanks @keqiaozhang

**Comment by marc-hb** [comment]
Pattern matching fix submitted in #946

**Pull request merged**